File: tests/fixtures/typescript/bumpy_road.ts
function validateAndProcess(user: any, order: any): any {
    const result: any = { valid: true, warnings: [] };

    if (user.active) {
        if (user.verified) {
            if (!user.suspended) {
                result.userOk = true;
            } else {
                result.valid = false;
                result.warnings.push("suspended");
            }
        }
    }

    if (order.total > 0) {
        if (order.items) {
            if (order.items.every((i: any) => i.inStock)) {
                result.orderOk = true;
            } else {
                result.warnings.push("out_of_stock");
            }
        }
    }

    if (order.payment) {
        if (order.payment.verified) {
            if (order.payment.amount >= order.total) {
                result.paymentOk = true;
            } else {
                result.valid = false;
            }
        }
    }

    return result;
}

File: tests/fixtures/python/bumpy_road.py
"""Functions with multiple nested conditional chunks (bumpy road)."""


def validate_and_process(user, order):
    result = {"valid": True, "warnings": []}

    # First bump: user validation
    if user.active:
        if user.verified:
            if not user.suspended:
                result["user_ok"] = True
            else:
                result["valid"] = False
                result["warnings"].append("suspended")

    # Second bump: order validation
    if order.total > 0:
        if order.items:
            if all(i.in_stock for i in order.items):
                result["order_ok"] = True
            else:
                result["warnings"].append("out_of_stock")

    # Third bump: payment check
    if order.payment:
        if order.payment.verified:
            if order.payment.amount >= order.total:
                result["payment_ok"] = True
            else:
                result["valid"] = False

    return result
